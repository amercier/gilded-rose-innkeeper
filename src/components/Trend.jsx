import React from 'react';
import { number, object } from 'prop-types';
import { Icon } from 'antd';

const colors = {
  up: '#52c41a',
  down: '#f5222d',
  stable: '#bfbfbf',
};

const icons = {
  up: 'caret-up',
  down: 'caret-down',
  stable: 'minus',
};

/**
 * Display a price and an indicator about its trend.
 *
 * @param {Object} props - React component properties.
 * @param {number} props.value - Trend value.
 * @param {Object} props.style - Additional style.
 * @returns {React.Element} The rendered element.
 */
const Trend = ({ value, style }) => {
  const trend = value === 0 ? 'stable' : value < 0 ? 'down' : 'up';
  return (
    <Icon type={icons[trend]} style={{ ...style, color: colors[trend] }} />
  );
};

Trend.propTypes = {
  value: number.isRequired,
  style: object,
};

Trend.defaultProps = {
  style: null,
};

export default Trend;
