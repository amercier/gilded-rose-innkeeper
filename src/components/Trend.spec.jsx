import React from 'react';
import { shallow } from 'enzyme';
import Trend from './Trend';

describe('Trend', () => {
  it('renders a grey minus for 0', () => {
    expect(shallow(<Trend value={0} />)).toMatchSnapshot();
  });

  it('renders a red up arrow for negative numbers', () => {
    expect(shallow(<Trend value={-1} />)).toMatchSnapshot();
    expect(shallow(<Trend value={-10} />)).toMatchSnapshot();
    expect(shallow(<Trend value={-123456} />)).toMatchSnapshot();
  });

  it('renders a green down arrow for positive numbers', () => {
    expect(shallow(<Trend value={1} />)).toMatchSnapshot();
    expect(shallow(<Trend value={10} />)).toMatchSnapshot();
    expect(shallow(<Trend value={123456} />)).toMatchSnapshot();
  });
});
