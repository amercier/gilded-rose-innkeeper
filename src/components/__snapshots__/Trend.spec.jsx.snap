// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trend renders a green down arrow for positive numbers 1`] = `
<Icon
  style={
    Object {
      "color": "#52c41a",
    }
  }
  type="caret-up"
/>
`;

exports[`Trend renders a green down arrow for positive numbers 2`] = `
<Icon
  style={
    Object {
      "color": "#52c41a",
    }
  }
  type="caret-up"
/>
`;

exports[`Trend renders a green down arrow for positive numbers 3`] = `
<Icon
  style={
    Object {
      "color": "#52c41a",
    }
  }
  type="caret-up"
/>
`;

exports[`Trend renders a grey minus for 0 1`] = `
<Icon
  style={
    Object {
      "color": "#bfbfbf",
    }
  }
  type="minus"
/>
`;

exports[`Trend renders a red up arrow for negative numbers 1`] = `
<Icon
  style={
    Object {
      "color": "#f5222d",
    }
  }
  type="caret-down"
/>
`;

exports[`Trend renders a red up arrow for negative numbers 2`] = `
<Icon
  style={
    Object {
      "color": "#f5222d",
    }
  }
  type="caret-down"
/>
`;

exports[`Trend renders a red up arrow for negative numbers 3`] = `
<Icon
  style={
    Object {
      "color": "#f5222d",
    }
  }
  type="caret-down"
/>
`;
